src/Draco.Compiler/Internal/Binding/Binder_Expression.cs

```diff
@@ -714,6 +726,23 @@ private async BindingTask<BoundExpression> BindIndexExpression(IndexExpressionSy
         }
     }
 
+    private BindingTask<BoundExpression> BindThisExpression(ThisExpressionSyntax syntax, ConstraintSolver constraints, DiagnosticBag diagnostics)
+    {
+        var function = ((SyntaxFunctionSymbol)this.ContainingSymbol!);
+        var thisArg = function.ThisArgument as ParameterSymbol;
```

Comment:
Avoid hard casting like this, you could use `as` and null-coalescing

Reply:
This is not a hard cast ?! this is as too ?! I need to do that in order to not specify the base type...

Reply:
`((SyntaxFunctionSymbol)this.ContainingSymbol!);`

Reply:
if only there was another way to specify the type :p

Reply:
The comment is on a _range_ of lines